src/doc/contrib/src/tests/profiling.md

```diff
@@ -2,15 +2,24 @@
 
 ## Internal profiler
 
-Cargo has a basic, hierarchical profiler built-in. The environment variable
-`CARGO_PROFILE` can be set to an integer which specifies how deep in the
-profile stack to print results for.
+Cargo leverages [tracing](https://crates.io/crates/tracing)
+as a basic, hierarchical built-in profiler.
 
-```sh
-# Output first three levels of profiling info
-CARGO_PROFILE=3 cargo generate-lockfile
+Environment variables:
+- `CARGO_LOG_PROFILE=<true|1>`: log tracing events to a file in the current working directory
+- `CARGO_LOG_PROFILE_CAPTURE_ARGS=<true|1>`: include arguments in the events
```

Comment:
Inspired by us previously having `CARGO_PROFILE` but not really expecting compatibility on it, I've removed the `__` prefixes but added a note below.

I kept the name as `CARGO_LOG_PROFILE` rather than switching to `CARGO_PROFILE` to associate this with `CARGO_LOG` in people's minds as that is the source for our profiling information.  Really, this is for `CARGO_LOG` visualization, and not just profiling, but I feel like current name is "good enough" rather than finding a "perfect" name that captures the uses for this.

Reply:
Sounds good. If we have `[log]` table in config in the future, we can just rename this.